### DISCUSSION TITLE
Is there support for Vendor / Specification Extensions?

### DISCUSSION THREAD

**Discussion by neongrau** [rswag/rswag#608] [General]
Trying to figure out how to add one or more `x-` prefixed info per verb to supply info about e.g. special authorizations required (for create / delete compared to show or index) and possible differences in results depending on access level.

Is this implemented yet?

**Comment by romanblanco** [comment]
@neongrau, you can send the `X-...` vendor identity in header by using 

```ruby
parameter name: :'X-VENDOR-IDENTITY', in: :header, schema: { type: :string }
```

**Reply by neongrau** [reply]
Not really what i meant. Swagger supports "Specification Extensions" or formerly called "Vendor Extensions".
These are required to have a name starting with `x-` but they aren't headers, they are just custom information added to the swagger.json generated and not visible via standard Swagger UI.

They are used for details that your own API logic uses. For example the index endpoints for every resource allows filters on a specific set of fields. Those i added into the descriptions for the parameters in markdown. But i was also asked to provide a machine readable Array of the supported fields so the Frontend can just read the swagger specs and knows it can offer filters on the fields listed.

Found a way to do it, probably not the intended way but exactly does what i needed and it seems to be valid OpenAPI.
See https://swagger.io/docs/specification/openapi-extensions/

```ruby
metadata[:operation]['x-gridfilter-fields'] = %w(title description created_at updated_at)
```
